src/xcode/ENA/ENA/Source/Scenes/CreateAntigenTestProfile/CreateAntigenTestProfileViewController.swift

```diff
@@ -49,15 +118,290 @@ class CreateAntigenTestProfileViewController: UIViewController, FooterViewHandli
 	func wasAttemptedToBeDismissed() {
 		dismiss()
 	}
+	
+	// MARK: - Protocol UITextFieldDelegate
 
-	// MARK: - Public
-
-	// MARK: - Internal
+	func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+		if textField.returnKeyType == .next {
+			if let cell = tableView.cellForRow(at: IndexPath(row: 1, section: 0)) as? InputCell, let currentIndex = cell.textFields.firstIndex(of: textField), (currentIndex + 1) < cell.textFields.count {
+				cell.textFields[currentIndex + 1].becomeFirstResponder()
+			} else {
+				textField.resignFirstResponder()
+			}
+		} else {
+			textField.resignFirstResponder()
+		}
+		return true
+	}
 
 	// MARK: - Private
 
 	private let viewModel: CreateAntigenTestProfileViewModel
 	private let didTapSave: () -> Void
 	private let dismiss: () -> Void
+	
+	private var dateFormatter: DateFormatter!
+	private var cancellables = [OpenCombine.AnyCancellable]()
+
+	private func setupView() {
+		// navigationItem
+		parent?.navigationItem.title = AppStrings.AntigenProfile.Create.title
+		parent?.navigationItem.rightBarButtonItem = dismissHandlingCloseBarButton
+		// view
+		view.backgroundColor = .enaColor(for: .background)
+		// tableView
+		tableView.separatorStyle = .none
+		tableView.register(DescriptionCell.self, forCellReuseIdentifier: DescriptionCell.identifier)
+		tableView.register(InputCell.self, forCellReuseIdentifier: InputCell.identifier)
+		tableView.keyboardDismissMode = .interactive
+		// date
+		dateFormatter = DateFormatter()
+		dateFormatter.dateStyle = .medium
+		dateFormatter.timeStyle = .none
+		dateFormatter.locale = Locale(identifier: "de_DE")
+	}
+	
+	private func setupBindings() {
+		viewModel.$antigenTestProfile
+			.sink { [weak self] antigenTestProfile in
+				let isSaveButtonEnabled =
+						!(antigenTestProfile.firstName?.isEmpty ?? true) ||
```

Comment:
Would expect the logic for enabling the button in the viewmodel. This way it also can be tested.